javascript-carousel/main.js

```diff
@@ -0,0 +1,108 @@
+// DOM queries and other variables
+var imageList = document.querySelectorAll('img');
+var buttonsList = document.querySelectorAll('.button-image');
+var timerID = null;
+
+// image change every so and so seconds
+function imageChangeWithInterval() {
+  imageChangeForward();
+  timerID = setTimeout(imageChangeWithInterval, 1300);
+}
+imageChangeWithInterval();
+
+// image changes going forward
+function imageChangeForward() {
+  var currentImage = document.querySelector('.current-pokemon');
+  var currentButton = document.querySelector('.current-button');
+
+  var nextImage = currentImage.nextElementSibling;
+  var nextButton = currentButton.nextElementSibling;
+
+  if (currentImage.id === 'jigglypuff' || currentButton.id === 'lastbutton') {
+    nextImage = document.querySelector('#bulbasaur');
+    nextButton = document.querySelector('#firstbutton');
+  }
+
+  currentImage.classList.remove('current-pokemon');
+  currentImage.classList.add('hidden');
+  nextImage.classList.remove('hidden');
+  nextImage.classList.add('current-pokemon');
+
+  currentButton.classList.remove('current-button');
+  nextButton.classList.add('current-button');
+}
+
+// click right arrow moves image reel forward
+function clickRightChevron(event) {
+  var rightArrow = document.querySelector('.fa-chevron-right');
+
+  if (event.target !== rightArrow) {
+    return;
+  }
```

Comment:
You have the right idea here using a guard to stop the function. But if possible, avoid querying the DOM during event handler functions. You can combine these lines by using [`Element.matches()`](https://developer.mozilla.org/en-US/docs/Web/API/Element/matches).

```js
if (!event.target.matches('.fa-chevron-right')) {
  return;
}
```